### PR TITLE
More operators

### PR DESCRIPTION
**Pull request by DonBraulio**
- [x] Refactor to ease organization of operators
- [x] Relational operators (binary/scalar versions): !=, >, >=, <=, <
- [x] Arithmetic ops (binary/scalar): power, modulo
- [x] Unary ops: inverse (~), isnan, notnan, abs, log
- [x] Implement logic ops (&, |)
- [x] Tests for everything 😅 
- [x] Add magic methods

### CODE REVIEW

**Review by ianspektor — APPROVED** [review]
👏🏼 laburazo! 

left a bunch of nitpicky comments, mostly about docstring style that I am trying to keep consistent throughout the lib :) implementation and tests look solid

File: temporian/core/data/node.py [hunk 1]
```diff
@@ -81,167 +81,331 @@ def __repr__(self) -> str:
...
 
+    def __bool__(self) -> bool:
+        # Called on "if node" conditions
+        # TODO: modify to similar numpy msg if we implement .any() or .all()
+        raise ValueError(
+            "The truth value of a node is ambiguous. Check condition"
+            " element-wise or use cast() operator to convert to boolean."
+        )
+
+    def __ne__(self, other: Any) -> Node:
```
[the remark this answers is not in the document]
Comment (DonBraulio) [reply temporian/core/data/node.py:92]: Yeah, I don't think it's ideal either. But we've to choose between being able to do `node1==node2` (i.e: like numpy) vs. the possibility to use nodes as dict keys (numpy arrays are not hashable).
Currently we use nodes as dict keys everywhere (i.e: evaluator), so it's not an option.

File: temporian/core/operators/binary/relational.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+    @property
+    def prefix(self) -> str:
+        return "lt"
+
+    # override parent dtype method
+    def output_feature_dtype(
+        self, feature_1: Feature, feature_2: Feature
+    ) -> DType:
+        return DType.BOOLEAN
```
[the remark this answers is not in the document]
Comment (DonBraulio) [reply temporian/core/operators/binary/relational.py:106]: Watch out for what's coming in the next commit

File: temporian/core/operators/unary.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+            ],
+            outputs=[pb.OperatorDef.Output(key="output")],
+        )
+
+    @classmethod
+    @property
+    @abstractmethod
+    def op_key_definition(cls) -> str:
+        """
+        Get the operator key used for serialization (build_op_definition)
```
[the remark this answers is not in the document]
Comment (DonBraulio) [reply temporian/core/operators/unary.py:85]: Fixed, thanks!

File: temporian/core/test/magic_methods_test.py [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+                Feature("f7", DType.INT32),
+                Feature("f8", DType.INT64),
+            ],
+            sampling=self.sampling,
+        )
+
+    def test_hash_map(self):
+        """
+        Test that the Node can be used as dict key
+        This is the reason to not overwrite __eq__ in Node
```
[the remark this answers is not in the document]
Comment (DonBraulio) [reply temporian/core/test/magic_methods_test.py:103]: Will do, wasn't aware of the style doc 😬

**Pull request merged**